packages/machine/src/protocol-types-tbd.ts

```diff
@@ -0,0 +1,50 @@
+// TODO: Probably merge this file with ./types.ts
+
+import { AppInterface, Terms } from "@counterfactual/types";
+import { BigNumber, Signature } from "ethers/utils";
+
+import { Protocol } from "./types";
+
+export type ProtocolMessage = {
+  protocol: Protocol;
+  multisigAddress: string;
+  params: ProtocolParameters;
+  fromAddress: string;
+  toAddress: string;
+  seq: number;
+  signature?: Signature;
+};
+
+export type SetupData = {};
+
+export type UpdateData = {
+  appInstanceId: string;
+  newState: object;
```

Comment:
tbh I don’t know what this type is I think this is unused and should be removed

Reply:
This is literally arbitrary state as defined by the developer writing their application — it can be anything

Reply:
Actually I see your point — it can be nested though

Reply:
maybe we can do

```
interface Json {
	[x: string]: string | number | boolean | Json | JsonArray;
}

interface JsonArray extends Array<string | number | boolean | Json | JsonArray> { }
```

IIRC there is a constraint similar to this anyway since `newState` is sent over the wire

modified from https://github.com/Microsoft/TypeScript/issues/1897#issuecomment-271146964